locallib.php

```diff
@@ -475,7 +477,29 @@ public static function coursedisplaytext($course, $tpl) {
             'IDNUMBER'  => $course->idnumber,
             'CATEGORY'  => $catname,
         );
+        // If we have limits defined, apply them.
+        static::apply_template_limits($replacements, $tpl);
         $displaytext = str_replace(array_keys($replacements), $replacements, $tpl);
         return strip_tags($displaytext);
     }
+
+    public static function apply_template_limits(&$replacements, &$template) {
```

Comment:
Please add a docblock for this function. You can use the "Moodle PHPdoc check" plugin to missing documentation blocks.